### ISSUE TITLE
Username with backslash no longer accepted

### ISSUE THREAD

**Issue by bpoland** [vmware/govmomi#1112]
Hi there,

I'm using an Active Directory user to authenticate to vSphere, so the username has a backslash in it (e.g. DOMAIN\username).  With govc 0.16.0 and earlier, I could set the GOVC_URL env variable with the backslash and it was happy.  For example:

`export GOVC_URL='DOMAIN\username:password@vsphere.server.fqdn'`

With govc 0.17.0 and .1, this same env variable doesn't work.  I get "govc: specify an ESX or vCenter URL" and when I manually specify the -u option I get this:

`govc: invalid value "DOMAIN\\username:password@vsphere.server.fqdn" for flag -u: parse https://DOMAIN\username:password@vsphere.server.fqdn: net/url: invalid userinfo`

I notice that the username right after "invalid value" in the above message has a double backslash, maybe some escaping behaviour has changed?  I took a quick look through the commit messages for changes from 0.16.0 to 0.17.0 and nothing jumped out at me.

I was playing around with it and I found a workaround -- if I put the URL-escape sequence for backslash (%5C) instead of the backslash, it does work.  But it would be nice not to have to do that.

Please let me know if I can provide any other info to help -- thanks!

**Comment by dougm** [comment]
Caused by this change to Go: https://github.com/golang/go/issues/23392

I suppose we could escape any `\` before calling `url.Parse()`?

Another workaround, avoids username/password in the call to `url.Parse()`:
```console
% export GOVC_USERNAME='DOMAIN\username' GOVC_PASSWORD='password' GOVC_URL='vsphere.server.fqdn'
```

**Comment by bpoland** [comment]
Ah, thanks! I didn't know you could separate out the username, password and URL like that.  I have verified that works, and I'm happy with that as a solution.  Maybe worth documenting somewhere for anyone else that runs into the same thing?

**Comment by dougm** [comment]
The vars and use case are mentioned here: https://github.com/vmware/govmomi/tree/master/govc#usage
But that was written before the mentioned change to Go.  We could add more detail on the issue, "invalid userinfo" error, etc.

**Comment by bpoland** [comment]
Thanks, I have suggested an update to that page here: https://github.com/vmware/govmomi/pull/1113

Also I wonder if it would be worth adding GOVC_USERNAME and GOVC_PASSWORD to the https://github.com/vmware/govmomi/blob/master/govc/USAGE.md page?  Or is that only for command line arguments?

**Comment by bpoland** [comment]
Actually I see that there's paperwork for me to sign to get a PR accepted -- maybe you can just make the change next time you're doing a commit :)

**Comment by dougm** [comment]
@bpoland should be just a few clicks for the CLA, would be great to have you contribute the change.  Otherwise, I tend to leave needs-doc labeled issues for taking care of just before creating a release.

**Comment by bpoland** [comment]
Yeah, there's a bunch of legalese in there that I'm not sure my employer would like me agreeing to...

Thanks again for your help with this!
